persist/fs/msgpack/roundtrip_test.go

```diff
@@ -101,6 +101,35 @@ func TestIndexInfoRoundtrip(t *testing.T) {
 	require.Equal(t, testIndexInfo, res)
 }
 
+// Make sure the new decoding code can handle the old file format
+func TestIndexInfoRoundTripBackwardsCompatibilityV1(t *testing.T) {
```

Comment:
Both these tests feel brittle because they rely on knowledge about the internals. Could you rely on only the public APIs for the encoder/decoder while testing compatibility (something like -  create a new file `persist/fs/msgpack/compatibility_test.go` and set the package to `msgpack_test`). Might need some changes to the encoder APIs to allow this.

Reply:
I could, but any changes I have to make to the msgpack encoder/decoder interfaces are gonna be sweeping.

I'm down to refactor this package in a separate P.R to have better support for encoding/decoding different versions at the public API level, I just think this P.R is already out of control and would rather not do it now considering I'd like to delete this code very soon

Reply:
How bout reworking so that the only code the internal code the tests use is a ctor to control the version? That'll localise the changes to be within the package, and make the tests a lot less brittle.

Reply:
Yeah thats a good compromise, I'll do that

Reply:
Cheers, the tests look much better now